java/arcs/core/util/UnionFind.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.util
+
+/**
+ * A union-find datastructure for computing equivalence classes
+ * consisting of `Element` instances.
+ */
+class UnionFind<Element> {
```

Comment:
Does this class need to be thread-safe? If not, it'd be good to call the fact that it's not thread-safe out in the KDoc.

Reply:
Initially, I don't think it has to be, but documented the class appropriately.